pyvista/core/common.py

```diff
@@ -320,7 +317,24 @@ def __init__(self, *args, **kwargs):
         self.references = []
         self._point_bool_array_names = []
         self._cell_bool_array_names = []
+        self._name = kwargs.pop('name', '')
+
+        # attempt to get the caller name (might be a geometric object)
+        if not self._name:
+            previous_frame = inspect.currentframe().f_back.f_back
+            function_name = inspect.getframeinfo(previous_frame)[2]
+            if function_name != '__init__':
+                self.name = function_name
```

Comment:
Huh, this is cool!

Unfortunately, the `function_name` is `wrap` for a ton of filter outputs and some geometry outputs like `pyvista.Cube()`:

```py
>>> import pyvista as pv
>>> pv.Cube()
PolyData (0x7fe072621468)
  Name      :	wrap
  N Cells   :	6
  N Points  :	24
  X Bounds  :	-5.000e-01, 5.000e-01
  Y Bounds  :	-5.000e-01, 5.000e-01
  Z Bounds  :	-5.000e-01, 5.000e-01
  N Arrays  :	2
```

and similarly for almost all filters:

```py
>>> pv.Sphere().clip()
PolyData (0x7fe07339f1c8)
  Name      :	wrap
  N Cells   :	980
  N Points  :	561
  X Bounds  :	-4.993e-01, 0.000e+00
  Y Bounds  :	-4.965e-01, 4.965e-01
  Z Bounds  :	-5.000e-01, 5.000e-01
  N Arrays  :	1
```

It would be really cool if we could set up filters to use the input mesh's name and then append the filter name. Perhaps this feature could come when I refactor the filters module.

In the meantime, it would be nice to not have `wrap` be the name so often

Reply:
Good point.  Let me see if I can figure out a simple fix without complicating the code base.